pkg/cmd/login/login.go

```diff
@@ -163,7 +162,15 @@ func runLogin(opts *Options) (err error) {
 			RedirectPath: build.MASSSORedirectPath,
 		}
 
-		if err = loginExec.Execute(context.Background(), ssoCfg, masSsoCfg); err != nil {
+		// Creating a global context with timeout
+		ctx, cancel := context.WithTimeout(context.Background(), build.DefaultLoginTimeout)
+		defer cancel()
+
+		if err = loginExec.Execute(ctx, ssoCfg, masSsoCfg); err != nil {
```

Comment:
```suggestion
		if err = loginExec.Execute(ctx, ssoCfg, masSsoCfg); err != nil && errors.Is(err, context.DeadlineExceeded) {
```

This could work, to save a nested if statement.